File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        seen = defaultdict(list)
        for i, v in enumerate(s):
            seen[v].append(i)
        
        for k in seen.values():
            if len(k) == 1:
                return k[0]
        return -1

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.java
class Solution {
    public int firstUniqChar(String s) {
         // Use a HashMap to store the indices of each character
        Map<Character, List<Integer>> seen = new LinkedHashMap<>();
        
        // Iterate over the string and store indices for each character
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            // If the character is not yet in the map, create a new list for it
            seen.putIfAbsent(c, new ArrayList<>());
            // Add the current index to the character's list of indices
            seen.get(c).add(i);
        }

        for(List<Integer> indices : seen.values()){
            if(indices.size() == 1){
                return indices.get(0);
            }
        }

        
        return -1;
    }
}